### ISSUE TITLE
[Other] About variable initialization using pointer

### ISSUE THREAD

**Issue by tyjeon24** [goldshakil/ElectricalProgrammingCourse2020#194]
Hello.

I opened this issue to inquire about how one memory containes two values at the same time; integer and its address.

I'm really confused to understand what *ptr and ptr actually do.

This is the code what I tested:
```c
#include <stdio.h>
main()
{
    int x, y, z;
    int* ptr = &z;
    *ptr = 1;

    x = *ptr;
    y = ptr;
}
```

The question is in x = *ptr; and y = ptr; part.
In assembly:
```assembly
    x = *ptr;
00E34E41  mov         eax,dword ptr [ebp-30h]  
00E34E44  mov         ecx,dword ptr [eax]  
00E34E46  mov         dword ptr [ebp-0Ch],ecx  
    y = ptr;
00E34E49  mov         eax,dword ptr [ebp-30h]  
00E34E4C  mov         dword ptr [ebp-18h],eax
```

Though there is a diffrence in using counter register, it seems equal assembly code to me. Both are moving the **ptr's value(ebp-30h)** to EAX register and move it to the x and y.

However, in fact, it is totally different. **x takes integer 1 while y takes the address of ptr**. It is obvious that x is 1 and y is address.

I would like to inquire how this could be possible despite the similar code.

Thank you for taking your time.

**Comment by goldshakil** [comment]
Very good question I will explain it today in a video but your interpretation of the assembly code is inaccurate.

> "one memory contains two values at the same time; integer and its address." 


The memory block can only hold a single value and it has  an address which is used to get that value. so what happens is as follows
Suppose that:
1) x is at address 0x1
2) y is at address 0x10
3) z is at address 0x100 and contains 1 value
4) ptr is at address 0x1000
now:
1) int *ptr=&z -> copies z's address into ptr variable
which means ptr is still at address 0x1000 but now it contains 0x100
2) x= * ptr -> copies the value pointed by the address stored at ptr variable
in other words, x=*(&0x100) 
thus x=1 since the value at address 0x100 is 1
3) y=ptr -> copies the value stored at ptr variable
y=0x100


the assembly code does the exact thing.
`00E34E41  mov         eax,dword ptr [ebp-30h]  `
here we move the value of ptr (0x100 in our example) to eax
`00E34E44  mov         ecx,dword ptr [eax]  `
then we move the value **pointed** by eax/0x100 to ecx which is 1 in our example.

I will explain today and upload a video it is really hard to explain like this
